azure-mgmt-trafficmanager/src/main/java/com/microsoft/azure/management/trafficmanager/TrafficManagerEndpoint.java

```diff
@@ -0,0 +1,500 @@
+     * Priority traffic-routing method
+     */
+    int routingPriority();
+
+    /**
+     * The entirety of a traffic manager profile endpoint definition as a part of parent definition.
+     *
+     * @param <ParentT> the return type of the final {@link Attachable#attach()}
+     */
+    interface Definition<ParentT> extends
+            DefinitionStages.Blank<ParentT>,
+            DefinitionStages.WithEndpointType<ParentT>,
+            DefinitionStages.WithEndpointThreshold<ParentT>,
+            DefinitionStages.WithSourceTrafficLocation<ParentT>,
+            DefinitionStages.WithAttach<ParentT> {
+    }
+
+    /**
+     * Grouping of traffic manager profile endpoint definition stages as a part of parent traffic manager
+     * profile definition.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of a traffic manager profile endpoint definition.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface Blank<ParentT> extends WithEndpointType<ParentT> {
+        }
+
+        /**
+         * The stage of the traffic manager profile endpoint definition allowing to specify the type.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithEndpointType<ParentT> {
+            /**
+             * Specify that the endpoint is an Azure cloud service, web app or public ip.
+             *
+             * @param resourceId the Azure resource id
+             * @return the next stage of the endpoint definition
+             */
+            WithAttach<ParentT> withTargetAzureResourceId(String resourceId);
+
+            /**
+             * Specify that the endpoint is an external endpoint that is not hosted in Azure.
+             *
+             * @param externalFqdn the external FQDN
+             * @return the next stage of the endpoint definition
+             */
+            WithSourceTrafficLocation<ParentT> withExternalFqdn(String externalFqdn);
+
+            /**
+             * Specify that the endpoint is a nested traffic manager profile.
+             *
+             * @param nestedProfile the nested traffic manager profile
+             * @return the next stage of the endpoint definition
+             */
+            WithEndpointThreshold<ParentT> withNestedProfile(TrafficManagerProfile nestedProfile);
+        }
+
+        /**
+         * The stage of the nested traffic manager profile endpoint definition allowing to specify the minimum
+         * endpoints to be online in the nested profile to consider it as not degraded.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithEndpointThreshold<ParentT> {
+            /**
+             * Specifies the child endpoint threshold.
+             *
+             * @param count the number of child endpoints to be online to consider nested profile as healthy
```

Comment:
minor, but the param wording is more explanatory than the function wording. Perhaps consider making the first one more explanatory: "Specifies the minimum number of endpoints to be online for the nested profile to be considered healthy."
and then the param could perhaps be: "a number of endpoints"